addons/sale_timesheet/models/sale_order.py

```diff
@@ -305,22 +325,34 @@ def _can_create_project(sol):
                     return True
             return False
 
+        def _needs_new_project(so_line, project):
+            """Rules are different based on the service_tracking:
+
+            - for 'project_only': create a new project if none created yet for this so_line
+            - for 'task_in_project': only create a new project if none configured on parent sale_order and if none created yet for this so_line"""
+
+            create_project_project_only = so_line.product_id.service_tracking == 'project_only' and not so_line.project_id
+            create_project_task_in_project = so_line.product_id.service_tracking == 'task_in_project' and not project
+
+            return create_project_project_only or create_project_task_in_project
+
         # task_global_project: create task in global project
         for so_line in so_line_task_global_project:
             if not so_line.task_id:
                 if map_sol_project.get(so_line.id):
                     so_line._timesheet_create_task(project=map_sol_project[so_line.id])
 
-        # project_only, task_new_project: create a new project, based or not on a template (1 per SO). May be create a task too.
+        # project_only, task_in_project: create a new project, based or not on a template (1 per SO). May be create a task too.
+        # if 'task_in_project' and project_id configured on SO, use that one instead
         for so_line in so_line_new_project:
-            project = so_line.project_id
-            if not project and _can_create_project(so_line):
+            project = so_line.project_id or so_line.order_id.project_id
+            if _needs_new_project(so_line, project) and _can_create_project(so_line):
```

Comment:
Maybe this method should returned the project and be called something like `_determine_project(so_line)`. Then the condition will check if no project and if it should create one.

Reply:
done